beacon-chain/sync/validate_beacon_attestation.go

```diff
@@ -84,6 +86,32 @@ func (s *Service) validateCommitteeIndexBeaconAttestation(ctx context.Context, p
 		return pubsub.ValidationReject
 	}
 
+	if featureconfig.Get().EnableSlasher {
```

Comment:
This skips a lot of DOS checks, you could potentially have peers feeding you junk which might overwhelm the node. If you can instead process the attestation 'later' then you could skip fetching the state twice.